javascript-view-swapping/main.js

```diff
@@ -0,0 +1,24 @@
+var tabContainer = document.querySelector('.tab-container');
+var tabElements = document.querySelectorAll('.tab');
+var views = document.querySelectorAll('.view');
+tabContainer.addEventListener('click', swap);
+function swap(event) {
+  if (event.target.matches('.tab')) {
+    for (var i = 0; i < tabElements.length; i++) {
+      if (tabElements[i] === event.target) {
+        tabElements[i].className = 'tab active';
+      } else {
+        tabElements[i].className = 'tab';
+      }
+
+    }
+  }
+  var attr = event.target.getAttribute('data-view');
+  for (var k = 0; k < views.length; k++) {
+    if (views[k].getAttribute('data-view') !== attr) {
+      views[k].className = 'view hidden';
+    } else {
+      views[k].className = 'view';
+    }
+  }
```

Comment:
Since this is outside of your conditional above, it will be done if the element clicked isn't a tab. If you click outside of the tabs, your views will likely all disappear. Move this code into your above conditional, and test that it works if you click the tab container but not a tab 👍